services/management/grpc/actions_server.go

```diff
@@ -260,38 +274,62 @@ func (s *actionsServer) StartMongoDBExplainAction(ctx context.Context, req *mana
 	}, nil
 }
 
-// StartPTSummaryAction starts pt-summary action.
+// StartPTSummaryAction starts pt-summary action. If the time since the last successful start of pt-summary action is lower than ptSummaryRefreshPeriod,
+// the response from the last action will be used. If the time is longer, the new pt-summary action will be called.
 //nolint:lll
-func (s *actionsServer) StartPTSummaryAction(ctx context.Context, req *managementpb.StartPTSummaryActionRequest) (*managementpb.StartPTSummaryActionResponse, error) {
-	agents, err := models.FindPMMAgentsRunningOnNode(s.db.Querier, req.NodeId)
+func (s *actionsServer) StartPTSummaryAction(ctx context.Context, psReq *managementpb.StartPTSummaryActionRequest) (*managementpb.StartPTSummaryActionResponse, error) {
+	// Gets current timestamp
+	timeNow := time.Now().Unix()
+
+	// Gets pointers to agents running on the node
+	psAgents, err := models.FindPMMAgentsRunningOnNode(s.db.Querier, psReq.NodeId)
 	if err != nil {
 		return nil, status.Errorf(codes.NotFound, "No pmm-agent running on this node")
 	}
 
-	agents = models.FindPMMAgentsForVersion(s.l, agents, pmmAgent2100)
-	if len(agents) == 0 {
+	// Filter by the version
+	psAgents = models.FindPMMAgentsForVersion(s.l, psAgents, pmmAgent2100)
+
+	// No agent found
+	if len(psAgents) == 0 {
 		return nil, status.Error(codes.NotFound, "all available agents are outdated")
 	}
 
-	agentID, err := models.FindPmmAgentIDToRunAction(req.PmmAgentId, agents)
+	// Gets the agent by ID
+	agentID, err := models.FindPmmAgentIDToRunAction(psReq.PmmAgentId, psAgents)
 	if err != nil {
 		return nil, err
 	}
 
-	res, err := models.CreateActionResult(s.db.Querier, agentID)
+	// If founds an old action record for the agent
+	if sAction, bFound := dicPtSummaryLastAction[agentID]; bFound {
```

Comment:
looks too complicated

Reply:
For comparison, I think a comment describing _what_ we do there would be nice – as Nurlan mention, it is somewhat complicated. But that should be a single comment describing the whole block, not Go-to-English translation of each line

Reply:
I can make a large comment before the if block. But my experience is that people skip long texts. That's why I tend to put a one-line comment before every action. What would you prefer as a recipe to cook something: Bullet form of ingredients or a paragraph of text?